### DISCUSSION TITLE
Benefits of the C API compared to the HTTP / GRPC protocols ?

### DISCUSSION THREAD

**Discussion by MatthieuToulemont** [triton-inference-server/server#6032] [Q&A] [answered]
Hello, 

While reading the documentation I stumbled upon the C API, is there any performance benefits in using it over the HTTP/GRPC protocols ? 

Best,

**Comment by nnshah1** [comment, marked as answer]
Yes - with the C API you can essentially embed Triton within your application and remove any overhead from the HTTP / GRPC protocols. In essence you create your own Triton based serving application. The HTTP / GRPC server components of Triton also use the C API internally.
